sourced/ml/transformers/indexer.py

```diff
@@ -12,14 +14,18 @@ class Indexer(Transformer):
     enumerating. Use value_to_index or [] to get index value.
     """
     def __init__(self, column: Union[int, str], column2id: Union[Dict[str, int], None]=None,
-                 **kwargs):
+                 cached_index_path: str=None, **kwargs):
         """
         :param column: column index or its name in pyspark.RDD for indexing.
         :param kwargs:
         """
         super().__init__(**kwargs)
         self.column = column
         self._value_to_index = column2id
+        self.cached_index_path = cached_index_path
+        if self.cached_index_path:
+            self._log.info("Loading the index from %s", self.cached_index_path)
+            self._value_to_index = DocumentFrequencies().load(source=self.cached_index_path).df
```

Comment:
You can remove `.df` and everything is likely to continue to work. This also means that you can remove the new method in the model.

Reply:
oh right hadn't seen the getter, on it